internal/lefthook/run_command.go

```diff
@@ -0,0 +1,40 @@
+package lefthook
+
+import (
+	"bytes"
+	"io"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/creack/pty"
+
+	"github.com/evilmartians/lefthook/internal/log"
+)
+
+type WaitFunc func() error
+
+func RunCommand(runner string, cmdRoot string) (*bytes.Buffer, bool, error) {
+	command := exec.Command("sh", "-c", runner)
+	if cmdRoot != "" {
+		fullPath, _ := filepath.Abs(cmdRoot)
+		command.Dir = fullPath
+	}
+
+	return RunPlainCommand(command)
+}
+
+func RunPlainCommand(command *exec.Cmd) (*bytes.Buffer, bool, error) {
+	ptyOut, err := pty.Start(command)
+	if err != nil {
+		return nil, false, err
+	}
+	defer ptyOut.Close()
+
+	commandOutput := bytes.NewBuffer(make([]byte, 0))
+	_, err = io.Copy(commandOutput, ptyOut)
+	if err != nil {
+		log.Debug(err)
+	}
```

Comment:
This should probably be a goroutine as the copying won't stop until `ptyOut` is closed

Reply:
Copying will stop when reader receive EOF. Do you think there might by some problems with that?

Reply:
AFAIK `ptyOut` won't return EOF unless it is closed. Not sure though. Gorouitine won't hurt anyone anyway I believe.

Reply:
It will, actually. Because we modify a commandOutput and also return it in main flow.

Reply:
Damn, yeah, you're right. On the other hand, we wait for the command completion and close the pipe on exit, so `commandOutput` shouldn't be modified after this function. Yet it's a pretty thin edge.

Also, I found this https://github.com/creack/pty/issues/127. It seems like tty is closed too early, so there may be problems with long-running processes.

However, I checked the code in master and it looks like everything should work like this